### PR TITLE
Patch `vsphere-cloud-controller-manager` daemonset image

### PR DESCRIPTION
**Pull request by g-gaston**
*Description of changes:*
Update `vsphere-cloud-controller-manager` image manually after upgrades
Unfortunately, the capv controller doesn't do this even if we update the `VSphereCLuster`, so we need to do it manually for now
In new versions of the capv provider, this is not managed by the controller directly anymore but just with a `ClusterResourceSet`
Which means that adding update capabilities to the `ClusterResourceSe` controller and updating our capv provider version will solve this problem

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/546#" title="Author self-approved">g-gaston</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [g-gaston]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/providers/vsphere/vsphere.go
Comment (taneyland) [inline pkg/providers/vsphere/vsphere.go:1639]: nit: typo in vsphere

**Review by taneyland — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/546#" title="Author self-approved">g-gaston</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/546#pullrequestreview-795631192" title="Approved">taneyland</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [g-gaston,taneyland]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
